### PR TITLE
[FIX] use feature-testing macro for std::type_identity in type_traits

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

**Comment by wvandertoorn** [comment]
travis is unhappy about a type that uses the anonymous namespace [-Werror=subobject-linkage] in the `instatiate` struct in core/type_traits/lazy.hpp:49

**Review by wvandertoorn — CHANGES_REQUESTED** [review]
Travis is unhappy about a type that uses the anonymous namespace [-Werror=subobject-linkage] in the `instatiate` struct in core/type_traits/lazy.hpp:49

What is the gain for using an anonymous namespace here, what does it fix?

**Comment by eseiler** [comment]
> What is the gain for using an anonymous namespace here, what does it fix?

Basically the same as `static`, but you can use it for everything. E.g. you can't to a `stattic` type.

File: include/seqan3/std/type_traits [hunk 1]
```diff
@@ -44,5 +46,6 @@ struct type_identity<t>
 //!\ingroup std_type_traits
 template <typename t>
 using type_identity_t = typename type_identity<t>::type;
+}
```
Comment (rrahn) [inline include/seqan3/std/type_traits:49]:
```suggestion
} // namespace
```

maybe?

**Comment by marehr** [comment]
I used the proper approach and fixed it via the feature testing macro; clang currently misses that macro, hence I created a bug ticket https://bugs.llvm.org/show_bug.cgi?id=47261

**Review by smehringer — APPROVED** [review]

**Pull request merged**